extensions/wikia/CuratedContent/CuratedContentSpecialController.class.php

```diff
@@ -187,65 +160,38 @@ public function save() {
 		}
 		$this->response->setFormat( 'json' );
 
-		$sections = $this->request->getArray( 'sections' );
-		list( $sections, $err ) = $this->processSaveLogic( $sections );
+		$helper = new CuratedContentHelper();
 
-		if ( !empty( $err ) ) {
-			$this->response->setVal( 'error', $err );
-			return true;
-		}
+		$sections = $helper->processLogic( $this->request->getArray( 'sections' ) );
```

Comment:
does getArray fallback to empty array or null when nothing is passed in the request? If it's null you can add fallback inside getArray method. This way we won't throw PHP errors about suppling null to foreach